src/Pelagos/Bundle/AppBundle/Security/LoginFormAuthenticator.php

```diff
@@ -114,12 +144,26 @@ public function getUser($credentials, UserProviderInterface $userProvider)
      * @param array         $credentials Credentials Array.
      * @param UserInterface $user        The user.
      *
+     * @throws AuthenticationException When account is locked out.
+     * @throws AuthenticationException When this is a bad password.
+     *
      * @return bool True if the credentials are valid.
      */
     public function checkCredentials($credentials, UserInterface $user)
     {
+        // Here check to see if $user is locked out?
+        if ($user->isLockedOut()) {
+            throw new AuthenticationException('Too many login attempts');
```

Comment:
Perhaps it would be better to lie here and just say "invalid credentials" like below. If we don't, we indirectly leak whether or not this account exists. I attempted to log in with 'rjones' (not a user) more than the prescribed number of times and never got this message.  It's just something to be aware of.